### PR TITLE
[1.3.6-prepare][FIX-#4617][LoggerServer] task log can not refresh in …

### PR DESCRIPTION
**Pull request by CalvinKirs**
…time #4618

issue #4617
pr #4618

### CODE REVIEW

**Comment by CalvinKirs** [comment]
@597365581 please take a look.Thx

**Comment by 597365581** [comment]
.ok

**Comment by 597365581** [comment]
okokok

**Review by lenboo — APPROVED** [review]
+1

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java [hunk 1]
```diff
@@ -349,28 +352,40 @@ private void clear() {
...
-                    long lastFlushTime = System.currentTimeMillis();
-
-                    while ((line = inReader.readLine()) != null) {
+        ExecutorService getOutputLogService = ThreadUtils.newDaemonSingleThreadExecutor(threadLoggerInfoName + "-" + "getOutputLogService");
+        getOutputLogService.submit(() -> {
+            BufferedReader inReader = null;
+            try {
+                inReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+                String line;
+                logBuffer.add("welcome to use bigdata scheduling system...");
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java]: I think this line should be removed

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java [hunk 2]
```diff
@@ -349,28 +352,40 @@ private void clear() {
...
+                while ((line = inReader.readLine()) != null) {
                         logBuffer.add(line);
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java]: The indent problem

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java [hunk 3]
```diff
@@ -349,28 +352,40 @@ private void clear() {
...
+        ExecutorService parseProcessOutputExecutorService = ThreadUtils.newDaemonSingleThreadExecutor(threadLoggerInfoName);
+        parseProcessOutputExecutorService.submit(() -> {
+            try {
+                long lastFlushTime = System.currentTimeMillis();
+                while (logBuffer.size() > 0 || !logOutputIsSuccess) {
+                    if (logBuffer.size() > 0) {
                         lastFlushTime = flush(lastFlushTime);
+                    } else {
+                        Thread.sleep(Constants.DEFAULT_LOG_FLUSH_INTERVAL);
+
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/AbstractCommandExecutor.java]: The empty line

**Review by chengshiwen — CHANGES_REQUESTED** [review]

**Review by chengshiwen — APPROVED** [review]
+1

**Review by davidzollo — APPROVED** [review]
+1
good job

**Pull request merged**